python/pyspark/sql/tests.py

```diff
@@ -3040,6 +3040,50 @@ def test_csv_sampling_ratio(self):
             .csv(rdd, samplingRatio=0.5).schema
         self.assertEquals(schema, StructType([StructField("_c0", IntegerType(), True)]))
 
+    def _get_content(self, content):
+        """
+        Strips leading spaces from content up to the first '|' in each line.
+        """
+        import re
+        pattern = re.compile(r'^ *\|', re.MULTILINE)
```

Comment:
We don't have to compile the pattern each time here since it's not going to be reused. You could just put this into re.sub I believe.

Reply:
Thanks! Fix it in 94f3414.